ArjunaCore/arjuna/tests/classes/com/hp/mwtests/ts/arjuna/recovery/RecoverySuspendTest.java

```diff
@@ -0,0 +1,159 @@
+import org.jboss.byteman.contrib.bmunit.BMUnitConfig;
+import org.jboss.byteman.contrib.bmunit.WithByteman;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+
+@WithByteman
+@BMUnitConfig(debug=true)
+@BMScript("recoverySuspend")
+public class RecoverySuspendTest {
+
+    private static RecoveryManager _manager;
+    private static RecordTypeMap _recordTypeMap;
+
+    // fields read with byteman
+    private final static int periodicRecoveryPeriod = 5;
+    private final static int recoveryBackoffPeriod = 1;
+
+    @BeforeAll
+    public static void beforeClass() {
+
+        RecoveryEnvironmentBean recoveryConfig = recoveryPropertyManager.getRecoveryEnvironmentBean();
+        String[] modules = {
+                // the module to test
+                "com.arjuna.ats.internal.arjuna.recovery.AtomicActionRecoveryModule"
+        };
+
+        // Let's go quick on this
+        recoveryConfig.setRecoveryBackoffPeriod(recoveryBackoffPeriod);
+        recoveryConfig.setPeriodicRecoveryPeriod(periodicRecoveryPeriod);
+        // don't sign off until the store is empty
+        recoveryConfig.setWaitForFinalRecovery(true);
+
+        // the test set of modules
+        recoveryConfig.setRecoveryModuleClassNames(Arrays.asList(modules));
+
+        // obtain a new RecoveryManager with the above config:
+        _manager = RecoveryManager.manager(RecoveryManager.INDIRECT_MANAGEMENT);
+        // recovery can start
+        _manager.startRecoveryManagerThread();
+
+        _recordTypeMap = new RecordTypeMap() {
+            @SuppressWarnings("unchecked")
+            public Class getRecordClass ()
+            {
+                return BytemanControlledRecord.class;
+            }
+
+            public int getType ()
+            {
+                return RecordType.USER_DEF_FIRST0;
+            }
+        };
+
+        RecordTypeManager.manager().add(_recordTypeMap);
+    }
+
+    @BeforeEach
+    public void beforeEach() {
+        // In case the Recovery Manager was suspended, this will resume it
+        _manager.resume();
+    }
+
+    @AfterAll
+    public static void afterClass() {
+        RecordTypeManager.manager().remove(_recordTypeMap);
+    }
+
+    @AfterEach
+    public void afterEach() {
+        if (!TxControl.isEnabled()) {
+            // Re-activate transaction creation
+            TxControl.enable();
+        }
+    }
+
+    @Test
+    public void testSuspensionWheneThereArentTxnsToRecover() {
+
+        // The Transaction System needs to be disabled, i.e. no new transactions can be created
+        TxControl.disable();
+
+        // Makes sure that the transaction reaper completed all transactions
+        TransactionReaper.transactionReaper().waitForAllTxnsToTerminate();
```

Comment:
I think this maybe should be `TransactionReaper.terminate(true)` (like in https://github.com/jmfinelli/narayana/pull/4/files#diff-7990bacf7d203544a582162b855619d55020e7bad28d66b6081822363250bec8R67)

Reply:
The Javadoc on the new method says we should only call waitForAllTxnsToTerminate when the transaction system has been shut down. Here you are wanting to use the method when we are suspending IIUC?

Reply:
Good point. Now that I re-read the javadoc (copied and pasted from other similar methods in `TransactionReaper`), the way I read/understood that note was that `TxControl.disable()` should be invoked in advance to make sure that the `the transaction system has been "shut down"` (i.e. shut down == disabled)

Reply:
I am no longer wanting to say TransactionReaper.terminate(true) so please feel free to resolve this thread (as a consequence of another comment, an implication is that this line would be removed entirely but even if that is not the decision I don't think I would ask for terminate anyway.

Reply:
Thanks Tom